### PR TITLE
Small fixes after testing.

### PR DESCRIPTION
**Pull request by andrewchambers**
I wasn't sure the extent that you are using mypy (or intend to use it), and am not very familiar with it myself.

These changes seemed to be enough for authentication to pass.

### CODE REVIEW

File: src/s3ql/backends/gs.py [hunk 1]
```diff
@@ -101,7 +101,11 @@ def __init__(self, ssl_context: ssl.SSLContext):
         self.conn_pool = dict()  # type: Dict[Tuple[str,int], HTTPConnection]
 
     def  __call__(self, url:str, method: str = 'GET',
-                  body : Optional[bytes] = None, headers=None, timeout=None):
+                  body : Union[bytes, str, None] = None, headers=None, timeout=None):
+
+        if type(body) == str:
+            body = cast(str, body)
+            body = body.encode('utf-8')
```
Comment (Nikratio) [inline src/s3ql/backends/gs.py]: Ugh, is this truly needed? S3QL is in no-position to decide what encoding the remote server expects.

File: src/s3ql/backends/gs.py [hunk 2]
```diff
@@ -529,7 +534,7 @@ def _get_access_token(self):
 
         if self.adc:
             try:
-                self.adc.refresh()
+                self.adc.refresh(GAuthHTTPRequestor(self.ssl_context))
```
Comment (Nikratio) [inline src/s3ql/backends/gs.py]: I would have hoped that the adc object saves the requestor instance - is this really not the case?

File: src/s3ql/backends/gs.py (hunk 1 above)
Comment (andrewchambers) [reply src/s3ql/backends/gs.py]: Yeah, I know ... I am not sure to why I was being passed a string from the google api making the request.

File: src/s3ql/backends/gs.py (hunk 2 above)
Comment (andrewchambers) [reply src/s3ql/backends/gs.py]: Passing None did not work In this case, I can check the code tomorrow.

File: src/s3ql/backends/gs.py (hunk 1 above)
Comment (andrewchambers) [reply src/s3ql/backends/gs.py]: @Nikratio The offending line is here https://github.com/googleapis/google-auth-library-python/blob/137b43ba002d422c68e7bfacdafa54986f38500b/google/oauth2/_client.py#L100

So If I understand correctly, all characters are ascii encoded.

The body object passed in comes from here:

https://github.com/googleapis/google-auth-library-python/blob/137b43ba002d422c68e7bfacdafa54986f38500b/google/oauth2/_client.py#L230

**Pull request merged**